### PR TITLE
feat(website): add official and HCP Packer Ready labels

### PR DESCRIPTION
**Pull request by zchsh**
👀 [Preview](https://packer-git-fork-zchsh-zslabel-built-in-plugins-deffcd-hashicorp.vercel.app)
🎟️ [Asana task](https://app.asana.com/0/1100423001970639/1201465687264559/f)

This PR adds an `Official` label to all of Packer's built-in plugins. It also adds an option for `HCP Packer Ready` labelling, both for built-in and external plugins.

## Official labels

`Official` labels have been added to all built-in plugins (to my limited, non-Packer knowledge) through the use of an MDX custom component placed at the top of each built-in plugin's `.mdx` file, like this:

```mdx
---
description: |
  The SSH communicator uses SSH to upload files, execute scripts, etc. on
  the machine being created.
page_title: Communicators - SSH
---

<BadgesHeader>
    <PluginBadge type="official" />
</BadgesHeader>

# SSH Communicator

Communicators are the mechanism Packer uses to upload files, execute scripts,
etc. on the machine being created, and are configured within the
[builder](/docs/templates/legacy_json_templates/builders) section.

...
```

Result:

CleanShot 2021-12-15 at 22 11 22@2x

These labels can be added on a per-document basis. Please do let me know if I've missed or incorrectly labelled any of these `.mdx` files, happy to update 🙇‍♂️

> Note on this approach: An alternate approach might be to try to automatically add these `"official"` labels to each built-in plugin's `.mdx` file. However, this would require relatively complex logic - for example, we do not want to add these labels to "Index" pages (such as [this one](https://www.packer.io/docs/post-processors)) or to the "Community Supported" list pages (such as [this one](https://www.packer.io/docs/post-processors/community-supported)). It felt simpler and more maintainable to add these badges manually. And perhaps most importantly, we would likely need to modify extract-transform-load workflows used for version docs in order to accommodate automatic build-time additions to docs files within this repo.

## HCP Packer Ready labels

`HCP Packer Ready` labels can now be added to any built-in or external plugin.

For built-in plugins, add (or append to) a `BadgesHeader` custom component in the `.mdx` file:

```mdx
---
description: |
  The HCP Packer Image Data Source retrieves information about an
  image from the HCP Packer registry. This information can be used to
  provide a source image to various Packer builders.
page_title: HCP Packer Image - Data Sources
---

<BadgesHeader>
    <PluginBadge type="official" />
    <PluginBadge type="hcp_packer_ready" />
</BadgesHeader>

# HCP Packer Image Data Source

Type: `hcp-packer-image`

The `HCP Packer Image` Data Source retrieves information about an
image from the HCP Packer registry. This information can be used to
provide a source image to various Packer builders.

...
```

Result:

CleanShot 2021-12-15 at 22 13 20@2x

For external plugins, add a `isPluginHcpPackerReady: true` property to the plugin's entry in `website/data/docs-remote-plugins.json`:

```json5
{
  /* ... */
  {
    "title": "HashiCups",
    "path": "hashicups",
    "repo": "hashicorp/packer-plugin-hashicups",
    "version": "latest",
    "isHcpPackerReady": true
  },
  /* ... */
}
```

Result:

CleanShot 2021-12-15 at 22 14 15@2x

## Other changes

This PR also fixes some minor technical debt related to the use of a `<br />` element for spacing, rather than CSS.

### CODE REVIEW

File: website/pages/docs/[[...page]].jsx [hunk 1]
```diff
@@ -10,7 +11,7 @@ import {
 } from 'components/remote-plugin-docs/server'
 
 //  Configure the docs path and remote plugin docs loading
-const additionalComponents = { PluginTierLabel, Checklist }
+const additionalComponents = { BadgesHeader, PluginTierLabel, Checklist }
```
Comment (zchsh) [inline website/pages/docs/[[...page]].jsx:14]: Allows `BadgesHeader` to be used in MDX.

File: website/content/docs/provisioners/powershell.mdx [hunk 2]
```diff
@@ -81,7 +83,7 @@ provisioner "powershell" {
   running on AWS, Azure, Google Compute, or OpenStack and would like to access
   the autogenerated password that Packer uses to connect to the instance via
   WinRM, you can use the `build` template engine to inject it using
-  ```{{ build `Password` }}```. In HCL templates, you can do the same thing by
+  `` {{ build `Password` }} ``. In HCL templates, you can do the same thing by
```
Comment (zchsh) [inline website/content/docs/provisioners/powershell.mdx:86]: This is strictly formatting, have confirmed these both render to `` {{ build `Password` }} `` (as seen on the [live page](https://www.packer.io/docs/provisioners/powershell)).

File: website/components/remote-plugin-docs/server.js [hunk 3]
```diff
@@ -52,10 +52,14 @@ async function generateStaticProps({
...
   // label that reflects the plugin tier
   // (current options are "Official" or "Community")
+  // and display whether the plugin is "HCP Packer Ready"
   function mdxContentHook(mdxContent) {
-    if (pluginTier) {
-      const tierMdx = `<br/><PluginTierLabel tier="${pluginTier}" />\n\n`
-      mdxContent = tierMdx + mdxContent
+    const badges = []
+    if (pluginTier) badges.push(pluginTier)
+    if (pluginIsHcpPackerReady) badges.push('hcp_packer_ready')
```
Comment (zchsh) [inline website/components/remote-plugin-docs/server.js:59]: Adds the `hcp_packer_ready` badge if flagged in the `docs-remote-plugins.json`.

File: website/components/plugin-tier-label/style.module.css [hunk 4]
```diff
@@ -40,11 +45,5 @@ to match Terraform Registry tier labels.
...
 .icon {
   margin-right: 4px;
-
-  & svg {
-    display: block;
-    & [fill] {
-      fill: var(--text-color);
-    }
-  }
+  color: var(--text-color);
```
Comment (zchsh) [inline website/components/plugin-tier-label/style.module.css:48]: Have refactored the `PluginTierLabel` component slightly - have instead used `BadgeHeader`, which leverages `PluginTierLabel` internally.

File: website/components/plugin-tier-label/index.tsx [hunk 5]
```diff
@@ -0,0 +1,42 @@
+import React from 'react'
+import InlineSvg from '@hashicorp/react-inline-svg'
+import svgRibbonIcon from './ribbon-icon.svg?include'
+import svgCheckIcon from './check-icon.svg?include'
+import classnames from 'classnames'
+import s from './style.module.css'
+
+type PluginTierType = 'official' | 'community' | 'hcp_packer_ready'
```
Comment (zchsh) [inline website/components/plugin-tier-label/index.tsx:8]: Have switched to Typescript in order to leverage this type in `BadgesHeader`.

File: website/components/badge/style.module.css [hunk 6]
```diff
@@ -1,10 +1,23 @@
-/*
-Note: colours have been hard-coded
-to match Terraform Registry tier labels.
-*/
 .root {
-  --background-color: #8f96a2;
-  --text-color: #fff;
+  /* theming. Note that gold and gray
+  have been set to hex values in order
+  to match Terraform Registry tier labels. */
```
Comment (jescalan) [inline website/components/badge/style.module.css:4]: 💯 attention to detail here

**Review by jescalan — APPROVED** [review]
🙌

File: website/data/docs-remote-plugins.json [hunk 7]
```diff
@@ -89,7 +89,8 @@
     "title": "HashiCups",
     "path": "hashicups",
     "repo": "hashicorp/packer-plugin-hashicups",
-    "version": "latest"
+    "version": "latest",
+    "isHcpPackerReady": true
```
Comment (nywilken) [inline website/data/docs-remote-plugins.json:93]: Thanks for calling out this setting for demo purposes. We can go in and add the details for the 5 plugins that are HCP Packer registry: vsphere, googlecompute, amazon, docker, azure. 

```suggestion
    "isHcpPackerReady": false
```

File: website/content/docs/communicators/ssh.mdx [hunk 8]
```diff
@@ -5,6 +5,10 @@ description: |
 page_title: Communicators - SSH
 ---
 
+<BadgesHeader>
+  <PluginBadge type="official" />
+</BadgesHeader>
+
```
Comment (nywilken) [inline website/content/docs/communicators/ssh.mdx:11]: I think it is okay not to label the communicators. 
```suggestion

```

File: website/content/docs/communicators/winrm.mdx [hunk 9]
```diff
@@ -5,6 +5,10 @@ description: |
 page_title: Communicators - Templates
 ---
 
+<BadgesHeader>
+  <PluginBadge type="official" />
+</BadgesHeader>
+
```
Comment (nywilken) [inline website/content/docs/communicators/winrm.mdx:11]: I think it is okay not to label the communicators. 
```suggestion

```

**Review by nywilken — APPROVED** [review]
I left a few suggestions, but this is otherwise solid.

**Comment by azr** [comment]
Nice ! 🤩

**Pull request merged**